src/rpc/methods/eth.rs

```diff
@@ -2252,9 +2261,214 @@ impl RpcMethod<1> for EthGetTransactionReceipt {
     }
 }
 
+#[derive(Debug)]
+pub struct CollectedEvent {
+    entries: Vec<EventEntry>,
+    emitter_addr: crate::shim::address::Address,
+    pub event_idx: u64,
+    reverted: bool,
+    height: ChainEpoch,
+    tipset_key: TipsetKey,
+    msg_idx: u64,
+    msg_cid: Cid,
+}
+
+fn match_key(key: &str) -> Option<usize> {
+    match key.get(0..2) {
+        Some("t1") => Some(0),
+        Some("t2") => Some(1),
+        Some("t3") => Some(2),
+        Some("t4") => Some(3),
+        _ => None,
+    }
+}
```

Comment:
Will this work on mainnet?

Reply:
Yes. `"t1"`, `"t2"`, etc, all refer to topics. Those are hardcoded keys: https://github.com/filecoin-project/builtin-actors/blob/master/actors/evm/src/interpreter/instructions/log_event.rs#L15